sky/resources.py

```diff
@@ -1885,16 +1975,39 @@ def _override_resources(
                     'Cannot specify both "any_of" and "ordered" in resources.')
 
         # Parse resources.accelerators field.
-        accelerators = config.get('accelerators')
-        if config and accelerators is not None:
-            if isinstance(accelerators, str):
-                accelerators = {accelerators}
-            elif isinstance(accelerators, dict):
-                accelerators = [
+        accelerators_original = config.get('accelerators')
+        accelerators = accelerators_original
+        if config and accelerators_original is not None:
+            if isinstance(accelerators_original, str):
+                accelerators_list = cls._parse_accelerators_from_str(
+                    accelerators_original)
+            elif isinstance(accelerators_original, dict):
+                accelerator_names = [
                     f'{k}:{v}' if v is not None else f'{k}'
-                    for k, v in accelerators.items()
+                    for k, v in accelerators_original.items()
                 ]
-                accelerators = set(accelerators)
+                accelerators_list = []
+                for accel_name in accelerator_names:
+                    parsed_accels = cls._parse_accelerators_from_str(accel_name)
+                    accelerators_list.extend(parsed_accels)
+            elif isinstance(accelerators_original, list) or isinstance(
+                    accelerators_original, set):
+                accelerators_list = []
+                for accel_name in accelerators_original:
+                    parsed_accels = cls._parse_accelerators_from_str(accel_name)
+                    accelerators_list.extend(parsed_accels)
+            # now that accelerators is a list, we need to decide which to
+            # include in the final set
+            accel_dict: Dict[str, bool] = {}
```

Comment:
Suggested rename: `accel_to_user_specified`. Please add a comment explaining what this logic is doing overall - determining for each accelerator whether it was directly specified by the user or was inferred from a manufacturer/memory spec.